core/src/main/java/org/mule/runtime/core/internal/processor/strategy/ProactorStreamEmitterProcessingStrategyFactory.java

```diff
@@ -133,12 +133,12 @@ public ProactorStreamEmitterProcessingStrategy(Supplier<Scheduler> ringBufferSch
     public Sink createSink(FlowConstruct flowConstruct, ReactiveProcessor function) {
       final long shutdownTimeout = flowConstruct.getMuleContext().getConfiguration().getShutdownTimeout();
       List<ReactorSink<CoreEvent>> sinks = new ArrayList<>();
-
       int concurrency = maxConcurrency < subscribers ? maxConcurrency : subscribers;
```

Comment:
doesn't this give the same result as `getParallelism()`?

Reply:
No. Subscribers, maxConcurrency and Parallelism are 3 different values. In the factory, the value given for parallelism is max(CORES, maxConcurrency), so it's at least as maxConcurrency. 
After trying different configurations this was the one that got best results in perf tests (and the one that takes into account maxConcurrency... if not, even if the user sets maxConcurrency = 1, the value won't be taken into account)